src/libAtomVM/nifs.c

```diff
@@ -5119,6 +5125,99 @@ static term nif_unicode_characters_to_binary(Context *ctx, int argc, term argv[]
     return result_tuple;
 }
 
+static term nif_erlang_lists_subtract(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc)
+
+    term list1 = argv[0];
+    term list2 = argv[1];
+
+    if (!term_is_list(list1) || !term_is_list(list2)) {
+        return BADARG_ATOM;
+    }
+
+    int proper;
+    int len = term_list_length(list1, &proper);
+    if (UNLIKELY(!proper)) {
+        RAISE_ERROR(BADARG_ATOM);
+    }
+
+    int proper2;
+    term_list_length(list2, &proper2);
+    if (UNLIKELY(!proper2)) {
+        RAISE_ERROR(BADARG_ATOM);
+    }
+
+    if (term_is_nil(list1)) {
+        return term_nil();
+    }
+
+    if (term_is_nil(list2)) {
+        return list1;
+    }
+
+    term *cons = malloc(len * sizeof(term));
+    if (UNLIKELY(IS_NULL_PTR(cons))) {
```

Comment:
`IS_NULL_PTR` is a shorthand for `UNLIKELY(cons == NULL)`, so no need to add an additional `UNLIKELY`.

Reply:
FWIW: I found that macro not bringing much value over manually expanding it.